mocca-client/src/test/java/com/paypal/mocca/client/MoccaSerializerTest.java

```diff
@@ -271,10 +296,47 @@ private Var newVar(String value, String... ignore) {
             @Override public String value() { return value;}
+            @Override public String[] ignore() { return ignore;}
+        };
+    }
+
+    @DataProvider(name = "ignoreSelectionSetCases")
+    public static Object[][] ignoreSelectionSetCases() {
+        return new Object[][]{
+                // Ignore one value of Complex type and 1 outer boolean var
+                {"{ \"query\" : \"query{getABeer(sampleRequest: {bar: \\\"bar\\\", foo: \\\"foo\\\"}) {complexField {innerBooleanVar innerIntVar} intVar stringVar}}\"}",
+                        "booleanVar", "complexField.innerStringVar"},
+                // ignore the entire complex Type
+                {"{ \"query\" : \"query{getABeer(sampleRequest: {bar: \\\"bar\\\", foo: \\\"foo\\\"}) {booleanVar intVar stringVar}}\"}",
+                        "complexField"},
+                // ignore everything
+                {"{ \"query\" : \"query{getABeer(sampleRequest: {bar: \\\"bar\\\", foo: \\\"foo\\\"}) {}}\"}",
+                        "complexField", "intVar", "stringVar", "booleanVar"},
+                // ignore nothing
+                {"{ \"query\" : \"query{getABeer(sampleRequest: {bar: \\\"bar\\\", foo: \\\"foo\\\"}) {booleanVar complexField {innerBooleanVar innerIntVar innerStringVar} intVar stringVar}}\"}",
+                        ""},
+                // ignore Complex type as a whole and also a field
+                {"{ \"query\" : \"query{getABeer(sampleRequest: {bar: \\\"bar\\\", foo: \\\"foo\\\"}) {booleanVar intVar stringVar}}\"}",
+                        "complexField.innerStringVar", "complexField"},
+                // ignore a field Not present in the response Type and an inner ComplexField value
+                {"{ \"query\" : \"query{getABeer(sampleRequest: {bar: \\\"bar\\\", foo: \\\"foo\\\"}) {booleanVar complexField {innerBooleanVar innerIntVar} intVar stringVar}}\"}",
+                        "complexField.innerStringVar", "complexField.someFieldNotPresent"}
+        };
+    }
+
+    @DataProvider(name = "valueWithIgnoreSelectionSetCases")
+    public static Object[][] ignoreWithValueSelectionSetCases() {
+        return new Object[][]{
+                // Custom Selection set with an Ignore value not present in it, Ignore attribute should not be used
+                {"{ \"query\" : \"query{getABeer(sampleRequest: {bar: \\\"bar\\\", foo: \\\"foo\\\"}) {booleanVar complexField {innerStringVar} intVar stringVar}}\"}",
+                        "{booleanVar complexField {innerStringVar} intVar stringVar}", "innerBooleanVar"},
+                // Custom Selection set with an Ignore value present in it, Ignore attribute should not be used
+                {"{ \"query\" : \"query{getABeer(sampleRequest: {bar: \\\"bar\\\", foo: \\\"foo\\\"}) {booleanVar complexField {innerStringVar} intVar stringVar}}\"}",
+                        "{booleanVar complexField {innerStringVar} intVar stringVar}", "complexField.innerStringVar"},
```

Comment:
Why `complexField.innerStringVar` was not ignored?

Reply:
These are the tescases in which we have value and ignore both set. 
When both are set Ignore attribute is not used and a warning is logged.